src/Runner/Runner.php

```diff
@@ -130,20 +130,28 @@ public function __construct(
     }
 
     /**
+     * @TODO consider to drop this method and make iterator parameter obligatory in constructor,
+     * more in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777/files#r1590447581
+     *
      * @param \Traversable<array-key, \SplFileInfo> $fileIterator
      */
     public function setFileIterator(iterable $fileIterator): void
     {
-        // @TODO consider to drop this method and make iterator parameter obligatory in constructor,
-        // more in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777/files#r1590447581
         $this->fileIterator = $fileIterator;
+
+        // Required only for main process (calculating workers count)
+        $this->fileCount = \count(iterator_to_array($fileIterator));
```

Comment:
for proper solution, I would consider to either fully remove `fileCount` var, or to keep that iterator_to_array result stored as `fileIterator`

Reply:
Wouldn't it lead to higher memory usage, since everything would be stored as an array? Right now we store the iterator, which only has the definition of a set, and additionally we store the count for further calculations. Yeah, maybe it's not super elegant, but perf-wise it is probably better?

Reply:
that's why ideally we do not need the fileCount at all